actionmailer/lib/action_mailer/base.rb

```diff
@@ -973,10 +973,11 @@ def collect_responses_from_templates(headers)
         templates_name = headers[:template_name] || action_name
 
         each_template(Array(templates_path), templates_name).map do |template|
-          self.formats = [template.format]
+          format = template.format || self.formats.first
```

Comment:
I really don't understand what this fallback is for. If the template doesn't have a format how does the Action Mailer rendering have one?

Also looks odd that we essentially can end up wiping out any other formats when reassigning e.g. `self.formats = [self.formats.first]`

Reply:
@tenderlove also pointed out that that looks weird. This `format` locals is used for two different things in this block:
1. Assigned as `self.formats = [format]`
2. Used to determine the content type we return.

2 is necessary (and is tested for) and the fallback makes sense for it (ex. for `welcome.erb`), but I'm not sure why 1 exists. My guess is that it's being used to set the requested format when rendering the template, but I'm not sure that's necessary... and if it is can we use `render(template, format: format)` instead? Removing it doesn't cause any test failures 🤷‍♂️. I'll open a separate PR for its removal (since this PR isn't changing behaviour here).